### ISSUE TITLE
Mutli GPU takes 10 times longer than single GPU

### ISSUE THREAD

**Issue by donal1** [songlab-cal/tape#86]
Hi guys, 

I've been trying to replicate the transformer results on pfam. I'm working with 2 Titan xp with 12 gb of vram. So I need a Batch size to gradient accumulation steps of 2.8 to 1 to prevent a cuda memory error. When I use a single gpu it takes roughly 14 secs per 20 iterations but with two gpus it takes 140 secs. It could be an error on my end put I replicted this on Google collab. So please let me know how I can reduce the training time with mutli gpu setup.

**Comment by rmrao** [comment]
So since the time TAPE was released, there have been a number of training frameworks released for pytorch. I have switched to using pytorch lightning and fairseq, depending on the project. DeepSpeed is also a good choice. It is difficult to maintain the framework for TAPE's training that was written before these frameworks were released, especially with continual updates to hardware and changes to the way pytorch distributed works. So I'd recommend trying to use one of these frameworks.

If you're specifically just trying to train transformer models, I'd strongly recommend either fairseq or DeepSpeed, which have excellent implementations for very fast multi-GPU training.
